index.js

```diff
@@ -12,6 +14,11 @@ module.exports = (input, needle, replacement) => {
 	let ret = '';
 	let matchCount = 0;
 	let prevIndex = 0;
+	const startIndex = opts.fromIndex > 0 ? opts.fromIndex : 0;
+
+	if (startIndex > input.length) {
+		return input;
+	}
 
 	while (true) { // eslint-disable-line no-constant-condition
 		const index = input.indexOf(needle, prevIndex);
```

Comment:
In my last commit, I left this [if statement](https://github.com/daveseco7/replace-string/blob/809c120ca6dae73a52701ee393d62bf2e55ba3dd/index.js#L18-L20). I can assume 0 if the fromIndex variable is greater then the string length, since a similar logic is being applied if fromIndex get a negative value. The main reason for the referred if statement is that it saves a while loop iteration.
I will let you decide on that.